lxd/instance/drivers/driver_lxc.go

```diff
@@ -2684,12 +2684,32 @@ func (d *lxc) Stop(stateful bool) error {
 	}
 
 	// Load the go-lxc struct
-	if d.expandedConfig["raw.lxc"] != "" {
+	lxcConfig, ok := d.expandedConfig["raw.lxc"]
+	if ok {
 		err = d.initLXC(true)
 		if err != nil {
 			op.Done(err)
 			return err
 		}
+
+		// Write to temp config file.
+		f, err := ioutil.TempFile("", "lxd_config_")
+		if err != nil {
+			return err
+		}
+
+		err = shared.WriteAll(f, []byte(lxcConfig))
+		f.Close()
+		defer os.Remove(f.Name())
+		if err != nil {
+			return err
+		}
+
+		// Load the config.
+		err = d.c.LoadConfigFile(f.Name())
```

Comment:
@monstermunchkin why cant we load the config file that is generated at start time and written to the log dir?